### DISCUSSION TITLE
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.

### DISCUSSION THREAD

**Discussion by darkdeathoriginal** [prisma/prisma#23741] [Q&A] [answered]
### Question
i have added `prisma generate` in postinstall section in package.json
i am getting this error in vercel.

```
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
--
20:30:51.257 | In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report
20:30:51.257 | at new PrismaClient (/vercel/path0/node_modules/.prisma/client/default.js:43:11)
20:30:51.257 | at 5333 (/vercel/path0/.next/server/app/api/auth/[...nextauth]/route.js:1:2376)
20:30:51.258 | at t (/vercel/path0/.next/server/webpack-runtime.js:1:128)
20:30:51.258 | at 82242 (/vercel/path0/.next/server/app/api/auth/[...nextauth]/route.js:1:1786)
20:30:51.258 | at t (/vercel/path0/.next/server/webpack-runtime.js:1:128)
20:30:51.258 | at 90522 (/vercel/path0/.next/server/app/api/auth/[...nextauth]/route.js:1:1194)
20:30:51.258 | at t (/vercel/path0/.next/server/webpack-runtime.js:1:128)
20:30:51.259 | at r (/vercel/path0/.next/server/app/api/auth/[...nextauth]/route.js:1:5749)
20:30:51.259 | at /vercel/path0/.next/server/app/api/auth/[...nextauth]/route.js:1:5780
20:30:51.259 | at t.X (/vercel/path0/.next/server/webpack-runtime.js:1:1196)
20:30:51.270 |  
20:30:51.271 | > Build error occurred
20:30:51.274 | Error: Failed to collect page data for /api/auth/[...nextauth]
20:30:51.274 | at /vercel/path0/node_modules/next/dist/build/utils.js:1258:15
20:30:51.274 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
20:30:51.275 | type: 'Error'
```

here is my nextauth options

```js
import prisma from "@/lib/prisma";
import CredentialsProvider from "next-auth/providers/credentials";

export const options = {
  providers: [
    CredentialsProvider({
      name: "Credentials",
      credentials: {
        username: {
          label: "Username",
          type: "text",
          placeholder: "Enter your username",
        },
        password: { label: "Password", type: "password" },
      },
      async authorize(credentials) {
        if (!credentials || !credentials.username || !credentials.password) {
          return null;
        }
        const dbUser = await prisma.user.findFirst({
          where: {
            username: credentials.username,
          },
        });
        if(dbUser && dbUser.password === credentials.password){
            const {password, ...user} = dbUser;
          return user;
        }
        return null;
      },
    }),
  ],
  callbacks:{
    async session({ session, token }) {
      session.user = token.user;
      return session;
    },
    async jwt({ token, user }) {
      if (user) {
        token.user = user;
      }
      return token;
    },
  }
  }

```

Do you know what could be causing the error.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by haena-j** [comment, marked as answer]
I also experienced the same error and, on a hunch, decided to remove the custom output. Surprisingly, that allowed for a successful deployment. 

before (schema.prisma)
```
generator client {
  provider = "prisma-client-js"
  output = "./client"
}
```
after (schema.prisma)
```
generator client {
  provider = "prisma-client-js"
}
```
If you're okay with not using the output, I suggest you also try to deploy after removing it.

Hoping to help resolve this issue, I'll share how I'm importing the Prisma Client.

src/lib/prisma.ts
```
import { PrismaClient } from "@prisma/client";

export const prisma = new PrismaClient();
```

src/app/api/hello/route.ts
```
import { prisma } from "@/lib/prisma";
```

**Reply by HtunHtun-Win** [reply]
thanks

**Comment by ludralph** [comment]
Hi @darkdeathoriginal 👋 

Are you generating the Prisma Client to a custom output location but importing from the default `@prisma/client`? You will need to modify the code to use the correct generated path instead of the default `@prisma/client`. Do give it a try and let us know if it works for you?

**Reply by darkdeathoriginal** [reply]
i was doing that after following this [answer](https://github.com/prisma/prisma/discussions/23741#discussioncomment-9041219) it got fixed.

**Comment by haena-j** [comment]
To provide a record and help others who might face a similar issue, I'd like to share my experience with configuring the Prisma Client.

Initially, I ran into a problem where the Prisma Client did not seem to generate correctly in the default "./node_modules/.prisma/client" path.

In seeking a solution, I turned to a specific discussion on the Prisma GitHub ([issue #2584](https://github.com/prisma/prisma/issues/2584#issuecomment-646414549)) and decided to manually set the output to its supposed default path.
schema.prisma:
```
generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/.prisma/client"
}
```
This workaround functioned perfectly in a local environment, allowing the Prisma Client to be imported using @prisma/client without any problems.

However, this manual output setting seemed to cause issues during deployment.
Ultimately, the solution was to remove the output configuration altogether and default back to the standard path, which resolved the deployment issues.

I've resolved my issue, so I'll wrap up here.!
